### PR TITLE
ci(cli): remove soroban CLI

### PR DESCRIPTION
**Pull request by jcs47**

### CODE REVIEW

File: .github/actions/setup-stellar/action.yaml [hunk 1]
```diff
@@ -29,12 +29,14 @@ runs:
...
-        curl -L -o soroban-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz https://github.com/stellar/stellar-cli/releases/download/v${{ inputs.stellar-version }}/soroban-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz
+        curl -L -o stellar-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz https://github.com/stellar/stellar-cli/releases/download/v${{ inputs.stellar-version }}/stellar-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz        
         tar -xvf stellar-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz -C stellar-binaries
-        tar -xvf soroban-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz -C stellar-binaries
         rm -rf stellar-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz
-        rm -rf soroban-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz
+
+        curl -L -o soroban-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz https://github.com/stellar/stellar-cli/releases/download/v${{ inputs.stellar-version }}/soroban-cli-${{ inputs.stellar-version }}-x86_64-unknown-linux-gnu.tar.gz
```
Comment (ahramy) [inline .github/actions/setup-stellar/action.yaml:36]: soroban-cli-*.tar.gz file is required to download, unzip and remove it? could we avoid downloading it?

File: .github/actions/compile-stellar/action.yaml [hunk 2]
```diff
@@ -37,9 +37,9 @@ runs:
       run: |
         mkdir -p ./stellar-binaries
         cp "$HOME/.cargo/bin/stellar" ./stellar-binaries/stellar
-        cp "$HOME/.cargo/bin/soroban" ./stellar-binaries/soroban
+        cp "$HOME/.cargo/bin/soroban" ./stellar-binaries/soroban | true
```
Comment (ahramy) [inline .github/actions/compile-stellar/action.yaml:40]: I guess we can just remove `soroban` cli completely

File: .github/actions/setup-stellar/action.yaml (hunk 1 above)
Comment (jcs47) [reply .github/actions/setup-stellar/action.yaml:36]: we could remove it entirely, but my idea is to allow the gh action to work with previous releases where the soroban client is still around

File: .github/actions/compile-stellar/action.yaml (hunk 2 above)
Comment (jcs47) [reply .github/actions/compile-stellar/action.yaml:40]: my idea was to account for previous releases that still include this cli

**Review by ahramy — APPROVED** [review]

**Pull request merged**